src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Publisher.kt

```diff
@@ -22,14 +22,18 @@ import com.google.cloud.pubsub.v1.Publisher
 import com.google.protobuf.ByteString
 import com.google.pubsub.v1.PubsubMessage
 import com.google.pubsub.v1.TopicName
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.FilenameUtils
 import org.slf4j.LoggerFactory
+import java.io.File
 
 
-fun getResultData(taskId: String, statusCode: StatusCode, resultBytes: ByteArray): ByteString {
+fun getResultData(taskId: String, statusCode: StatusCode, compiledFile: File): ByteString {
     return CompilerFleet.CompilerFleetResult.newBuilder()
             .setTaskId(taskId)
+            .setResultBytes(ByteString.copyFrom(FileUtils.readFileToByteArray(compiledFile)))
+            .setRootFileName(FilenameUtils.removeExtension(compiledFile.name) + PDF_EXTENSION)
```

Comment:
I don't think that PDF file name in the result makes too much sense. The client is probably aware of the issued task and its root file (by the way, it should check task id and process only those results which are produced in response to issued tasks).

Anyway, even if it makes any sense, it is either not root file name (because root file name is something different) or it should not have PDF exension. That is, you either pass result file name with PDF extension or the real root file name from the compile task to let client decide how to name the result file.

Reply:
Fixed. Added checking taskId in message processing